### ISSUE TITLE
Sprint - Dec 9 - Dec 20

### ISSUE THREAD

**Issue by pauldambra** [PostHog/posthog#26634]
# Global Sprint Planning

## 3 things that might take us down

1.
2.
3.

## Urgent incident follow-ups

https://github.com/orgs/PostHog/projects/103/views/2

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
# Team ___

**Support hero:** ___

## Retro

<!-- Grab the high and low priority items from last time and add whether that item was completed or not -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red


### High priority

-

### Low priority / side quests

-

```

**Comment by pauldambra** [comment]
# Team And They All Rolled Over And One Fell Out

xmas party sprint

**Support hero:** @pauldambra 

<details>
<summary>Retro</summary>

* @pauldambra safari bug was horrible, but at least fixed, good but tiring to make progress as well as fight bugs


## what we shipped

## Mobile

* Capture network headers and body (assuming new instrumentation is in place by then) https://github.com/PostHog/posthog-ios/issues/159 @ioannisj 
* 🪲 long sessions on mobile - https://github.com/PostHog/posthog-ios/issues/227.  @ioannisj 
   * still not sure what's causing this but quite a few reports 
* get people using iOS autocapture @ioannisj and @marandaneto 
   * ask for help from marketing? blog post? socials?
   * find customers that asked for it in the past?
* flutter replay alpha and getting feedback @marandaneto
* [react native](https://github.com/PostHog/posthog-js-lite/issues/240) person profiles @marandaneto 
 
## Replay

* ✅ replay 4000 low hanging fruit @pauldambra 
   * got a chunk of this done
   * meeting adam in london to talk ux

### Low priority / side quests

* Capture iOS logs for recordings https://github.com/PostHog/posthog-ios/issues/258 @ioannisj 
* Enforce max session duration https://github.com/PostHog/posthog-ios/issues/245  @ioannisj 
* 🔄 a brief diversion before querying network data - wrap websockets as well as fetch @pauldambra
* 🔄 (can we|do we)  measure watch time and buffer time? @pauldambra 
* 🔄 go through support and find people who wanted block list and event trigger @pauldambra 
* [android](https://github.com/PostHog/posthog-android/pull/199) min session duration (cost control) @marandaneto
    * waiting on external contributor 


</details>

<details>
<summary> OKRs 🟡 </summary>

1. OKR, status (red/yellow/green) and action points if yellow/red

🟡📱Goal 1: People think of PostHog as a mobile solution
🟡 📹 Goal 2: Replay 4000
    - this is now so excellent it's Replay 4000
    - https://github.com/PostHog/posthog/issues/26098
🟡 🎤 Goal3: Ship the things users keep asking for
 - [ ] queryable network data
 - [x] start replay on trigger e.g. event
 </details>

### High priority

## Mobile

## Replay

* a brief diversion before querying network data - wrap websockets as well as fetch @pauldambra

### Low priority / side quests

* (can we|do we)  measure watch time and buffer time? @pauldambra 
* go through support and find people who wanted block list and event trigger @pauldambra

**Comment by jurajmajerik** [comment]
# Team Experiments 🧪 

## Retro

## Sprint goals

**Comment by Phanatic** [comment]
## Team Surveys 

@Phanatic offboarding December 23rd. 

### Retro
✅ Shipped Adaptive sampling for survey responses. 
✅ Bug fixes for NPS recurring surveys. 
✅ Customer support 

### Sprint goals
* Fixing bugs reported by Annika and a few customers that we're getting more survey responses than allowed when using Adaptive Sampling.
* Fixing bugs reported by customers of NPS recurring surveys who are confused about when the survey is shown or hidden.
* Handing off no-code experiments to the Experiments team.
* Handing off Surveys to Manoel and Ioannis
* Customer support 
* Q1 Planning

**Comment by MichaelKutsch-ph** [comment]
# Team Infrastructure


## Retro



- 

## Hang over items from previous sprint



- Item 1. prioritised/deprioritise

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red

**Comment by benjackwhite** [comment]
# Team Conversion Drivers & Planners (CDP)

- Support: Marius

## Retro

- 🔴 @benjackwhite / @mariusandra  client->posthog->destination event mapping system for ad destinations + infra
  - Swapped to Marius' todo - not a huge amount of progress due to sickness + incident work
- 🟢 @mariusandra plumbing for allowing custom JS (GTM replacement anybody??)
  -  Pretty much there although waiting on the Mappings + stateless asset delivery to be production ready
- 🟡 @MarconLP building and testing client side destinations, E2E tests
  - E2E testing was a bust due to captchas etc so something to come back to in the future
  - Client side destinations weren't ready but major improvements to the Google Ads integration that is now basically there
- @benjackwhite also distracted with continous onboarding of Infra team as well as planning ahead to new hires
 
## OKRs
* 🟢  Widespread usage of the new CDP 
* 🔴 Messaging V1 
* 🟡 Ingestion is rock solid

### High priority
* @benjackwhite gradually roll out RemoteConfigs to everyone (providing the code for site-functions)
* @mariusandra finish Mappings work with focus on TikTok Ads
* @MarconLP build out and test TikTok Ads client side integration

### Side quests
* Ideas for growth hacking of destinations (big list of all known SaaS we could work with that people can search for to upvote etc.
* Continue investigations into Warpstream events capture (utilize return of Frank perhaps)

**Comment by timgl** [comment]
# Team Movers of Data

**Support hero:** Tomas/Ross

## Retro



- We managed to get to the bottom of missing events in batch exports
- Could have better alerts

## Hang over items from previous sprint



- Item 1. prioritised/deprioritise

## OKR

Coming soon!

### High priority

- #26317 - @rossgray 
- #26644 - @rossgray
- Use encrypted fields  - @tomasfarias 
- https://github.com/PostHog/posthog-cloud-infra/issues/3305 - @tomasfarias

**Comment by dmarticus** [comment]
# Team Flags

## Support

- 12-2 to 12-9 @dmarticus 
- 12-9 to 12-16 @havenbarnes 

## Retro

### Things we got done

- [x] Track `feature flag called` events in usage reports so we can track up who set up flags successfully @dmarticus 
- [x] generally ironing out service (centralizing error handling, reducing memory usage overhead with smarter copying/cloning)
    - found a bug in the following two paths (1) feature flag payloads, and (2) group targeting without group overrides.  PR to come later today.
- [x] land v0 of flag staleness feature @havenbarnes 
- [x] Fixed issue with feature flag payloads SDK methods on the server-side (Node, Python) not correctly sending `feature_flag_response` payloads @dmarticus 
- [x] Show feature flag payload as official PostHog property @havenbarnes
- [x] Adding some debugging event properties to $feature_flag_called to help indicate when SDKs are using bootstrapping  

### Things we didn't get done and why

- [ ] testing/productionizing new `/flags` service @dmarticus 
    - [ ] functional testing via django-compat integration tests
    - [ ] "real" testing via mirroring traffic to new service (meeting with ingestion team 11-20 to go through this.

As I was writing more integration tests, I came across some bugs, and needed to fix them.

- [ ] Start work on feature management UI @havenbarnes 
    - needed realignment on the correct UI abstractions for this feature; made an initial PR to start that conversation, and are iterating on it.

---

## Sprint

### High priority

- [ ] testing/productionizing new `/flags` service @dmarticus 
    - [ ] fixing P0 bugs along `/flags` hotpath 
    - [ ] functional testing via django-compat integration tests
    - [ ] "real" testing via mirroring traffic to new service

### Side Quests

- [ ] Continue to make sentry more actionable and useful @havenbarnes 
- [ ] Figure out how to cut off the flags bit of `/decide` (and, in future `/flags`) when folks go over – right now we don't stop serving flags when overage happens @dmarticus 
- [ ] Continue to look at growth metrics and see what other properties we can add to better understand our growth @dmarticus 

#### Things to chew on

- [ ] Can we know when people are starting to make new features?

## OKRs

1. Ship new feature flags in production and accomplish the 10x scale + 2x speed boost performance targets (ideally decreases infra spend by N% as well, will try to figure out a more specific target going forward)
2. Maintain current /decide service and make optimizations.

**Comment by EDsCODE** [comment]
# Team Data Storehouse
### OKR Q4 2024

### Objective

**Query 3000**

* **Key Results:**
    * Autocomplete
	* Increase general BI experience/product https://github.com/PostHog/meta/issues/157
	* Declutter the data warehouse UI and make the features intuitive to find

**Data Modeling MVP**

* **Key Results:**
	* Infrastructure decided and implemented
	* Integrating external data with feature flags
	* External data everywhere in insights/persons/cohorts
	* Get billing team to use modeling in posthog for their invoices_with_annual table
 
### Retro
Team split in 2

- [ ] Data import rebuild @Gilbert09 
	- release internally by end of week
- [ ] SQL editor 3000 finish MVP section + move to new priorities @EDsCODE 
	- release internally be end of week

### Priority
Tom off for most of sprint

- [ ] Draft queries + reorg sidebar (https://github.com/PostHog/posthog/issues/26193) @EDsCODE 
- [ ] Materialization Tab (https://github.com/PostHog/posthog/issues/26193) @EDsCODE 
- [ ] Permanent search bar
- [ ] Release data import to everyone @Gilbert09

**Comment by aspicer** [comment]
# Team Product Analytics

🎉 Peter Joining Dec 9th!

## Q4 2024 objectives

**Legacy 0x** - Thomas owning

- All insights are query-based
- rm -rf posthog/queries/

- Insight query is stored in the URL for instant sharing of unsaved insights

**First-class environments** - Michael owning

- Environments are rolled out to everyone

**Threshold-based alerts** - Anirudh owning

- All users get to set up alerts on PostHog metrics
- Alerts are launched with Marketing, and positioned into paid plans
- We're using alerts ourselves for key trends and funnels

**Flexible funnels** - Sandy owning

- UDF funnels rolled out to everyone, unlocking arbitrary complexity
- Allowing parts of funnels to be unordered. No more separation between regular, unordered, and strict funnels
- Optional funnel steps

**10x onboarding and ease of discovering features** - Anirudh and Sandy owning

- We run an exercise in implementing PostHog in a hobby project. What stands out as painful or unclear?

**User delight 2x** - support rotation owning

- Support is 1 hero + 1 sidekick
- Every week support folks ship those small requested features (see [[list of requests so far](https://docs.google.com/spreadsheets/d/1-lXjfzo6313Ai2vMf6smmoA0zghitBQTRD4lPgZ-XqA/edit?gid=0#gid=0)](https://docs.google.com/spreadsheets/d/1-lXjfzo6313Ai2vMf6smmoA0zghitBQTRD4lPgZ-XqA/edit?gid=0#gid=0))
- Goal: 90% of tickets fulfill the SLA

**B2B analytics research** - Anna owning

- Understand the market and user needs better in order to improve our current Group analytics product next quarter

## Retro

Anirudh (Support second week)

- 🟢 Alert Comms / Launch
    - Official docs out, monitoring
- 🟢 Support for launch / fast-follow features for alerts
- 🟡 Cached dashboards (https://github.com/PostHog/posthog/issues/26309)
    - Remainder of sprint - rolling to more customers

Sandy

- 🟢 Finish Cohort Migration
    - Passing local tests
    - Need to write testing framework for production data
- 🟡 Spec out funnel improvements for December https://github.com/PostHog/posthog/issues/17313
- 🟢 Plan Peter's Onboarding, find tasks
- [Side Quest] Hi-pri customer query issues

Anna

- 🟢 Group Analytics Research, generate plan for Q1
    - Survey sent out and user interviews in progress

Thomas (Support First Week)

- Migrate api endpoints to use hogql queries
- Look into filters removal broadly
- Fix lifecycle charts with personless events

## Upcoming Sprint

**Support hero:** @aspicer @thmsobrmlr 

**SDK Support hero**: @anirudhpillai @aspicer

- Alerts [Anirudh]
    - Scaling with the rollout
        - Improving execution system for running alerts
        - Bugfixes and deploy monitoring

 - Carryover: Cached Dashboards, further rollout [Anirudh]
    - Thinking about API concurrency and scaling

- Peter Onboarding [Sandy]

- Carryover: Cohort migration production monitoring and bug fixing [Sandy]

- Preparing for Planning [Anna]

    - Collect build better notes and user interviews for Q1 Planning

Thomas?
